candig_federation/api/federation.py

```diff
@@ -4,162 +4,275 @@
-            if peer != app.config["self"]:
-                uri_list.append(peer)
 
-        for future_response in self.async_requests(uri_list, self.request, header):
+        # Old logic when using "local" peers vs uniform handling
+
+        # for peer in APP.config["peers"].values():
+        #     if peer != APP.config["local"]:
+        #         uri_list.append("{}".format(peer))
+
+        for peer in APP.config["peers"].values():
+                uri_list.append("{}".format(peer))
+
+        print("Federating: {}".format(uri_list))
+        for future_response in self.async_requests(uri_list=uri_list,
+                                                   request_type=request,
+                                                   header=header,
+                                                   endpoint_payload=endpoint_payload,
+                                                   endpoint_path=endpoint_path):
             try:
                 response = future_response.result()
-            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
-                self.status.append(503)
+                print(response.status_code)
+            except AttributeError:
+                if isinstance(future_response, requests.exceptions.ConnectionError):
+                    self.status.append(404)
+                if isinstance(future_response, requests.exceptions.Timeout):
+                    self.status.append(408)
                 continue
+            except requests.exceptions.ConnectionError:
+                self.status.append(404)
+                continue
+            except requests.exceptions.Timeout:
+                self.status.append(408)
+                continue
+
             self.status.append(response.status_code)
             # If the call was successful append the results
 
             if response.status_code == 200:
                 try:
-                    if self.request == "GET":
-                        self.results.append(response.json()["results"])
-
-                    elif self.request == "POST":
-                        # peer_response = response.json()["results"]
-                        peer_response = response.json()
-
-                        if not self.results:
-                            self.results = peer_response
-                        else:
-                            # for entries in peer_response["data"]["datasets"]:
-                            #     print(self.results)
-                            #     self.results["data"]["datasets"].append(entries)
-                            self.results.append(peer_response)
+                    """
+                    Each Response will be in the form on a ResponseObject
+                        {"status": [], "results": []}
+                    Gather the data within each "results" and append it to 
+                    the main one.
+                    """
+                    self.results.append(response.json()["results"])
```

Comment:
All responses should look like this, but we should handle keyerrors, when for one reason or another "results" field does not exist

Reply:
If a key or value error shows up in this section of the code, where we're receiving a 200, should the program append a 500 instead along with some type of error message within "results"?

Reply:
It is more like a response format issue (in case we change the format, but it is not propagated on the network properly), than an internal server issue, hence I am not sure whether 500 is the best option, but sounds like a good enough solution.